### PR TITLE
Transformers as data

### PR DESCRIPTION
**Pull request by wpears**
Includes changes that moves the onus of transforming data from loader into retriever. The reasoning for this is that:
- it keeps metadata about the remote files together in one place, making updates easier
- Smaller files to upload / download by retriever / loader respectively
- Less space required on S3

This will also allow loader to be simpler, which will be captured in a later PR.

### CODE REVIEW

**Comment by hkeeler** [comment]
Does this impact open _loader_ PRs https://github.com/cfpb/grasshopper-loader/pull/93 and/or https://github.com/cfpb/grasshopper-loader/pull/96?

**Comment by wpears** [comment]
#93 can certainly be merged. You could hold off on #96 if you'd like, since the bulk of that is transformer related, but I looks to me like there are a few bug fixes in there as well, so merging it wouldn't hurt.

File: lib/fieldFilter.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    for(var i=0; i<fields.length; i++){
+      var field = fields[i];
+      var val;
+
+      if(field.type === 'dynamic'){
+        try{
+          val = new Function('props', field.value)(props); //eslint-disable-line
+        }catch(e){
+          return cb(null);
```
Comment (hkeeler) [inline lib/fieldFilter.js:27]: What type of exception could occur here?  Is it OK just return without the underlying exception or logging that something went arwy?

File: lib/fieldFilter.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        }
+      }else{
+        val = props[field.value];
+      }
+
+      if(!val) return cb(null);
```
Comment (hkeeler) [inline lib/fieldFilter.js:33]: Similar to above, wouldn't we want to know if a given field had no value?

**Comment by hkeeler** [comment]
All reviewed up.  Just a couple questions in there.

File: lib/fieldFilter.js (hunk 1 above)
Comment (wpears) [inline lib/fieldFilter.js:27]: Yeah, this is what happens now, part of what has been folded into the large, soon-to-be pull request. The original idea for just returning was that the data count at the end would be sufficient to show that some data was dropped... but pretty much always you'd want to see what that data is, even if it spams your logs

File: lib/fieldFilter.js (hunk 2 above)
Comment (wpears) [inline lib/fieldFilter.js:33]: This has also been adjusted.

**Comment by hkeeler** [comment]
Sounds good.  Look forward to the forthcoming mega-PR. :)  Merging...

**Pull request merged**